[Term Entry] C: Linked List (#4853)
Codecademy/docs: content/c/concepts/linked-list/linked-list.md

---
Title: 'Linked List'
Description: 'Linked list is a type of data structure in which elements are not sequentially stored but are randomly spread in memory.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Data Structures'
  - 'Linked Lists'
  - 'Pointers'
  - 'Memory'
  - 'Variables'
CatalogContent:
  - 'learn-c'
  - 'paths/computer-science'
---

> A **`linked list`** is a type of linear data structure in which elements are not sequentially stored but are randomly spread in memory consisting of small units called `node`. These units connect and point to other nodes to create a linked list. A node is the smallest unit of the linked list, responsible for storing two different types of values first is the value we are storing, which can be of any data type, and the second is a pointer used for storing the address of the next node.

Reviewer: ```suggestion
A **linked list** is a type of linear data structure in which elements are not stored sequentially but are dynamically allocated in memory, consisting of small units called nodes. These units connect and point to other nodes to create a linked list.
```

Reviewer: Removed the last line, because you've already discussed it in the below structure of a node.

> The structure of a node is shown in the below image.

Reviewer: ```suggestion
The structure of a node is illustrated below:
```

![Linked list node](https://raw.githubusercontent.com/Codecademy/docs/main/media/linked-list-node.png)

> So the node consist of two parts:

Reviewer: ```suggestion
Each node in a linked list has two parts:
```

> - `data`: The data that will be stored, this can be of any data type present in `C`.

Reviewer: ```suggestion
- `data`: It stores the value, which can be of any data type.
```

> - `next`: This is a `pointer` that is used to store the address of a different node.

Reviewer: ```suggestion
- `next`: A pointer that stores the memory address of the next node in the sequence.
```

> When nodes are connected to each other a linked list is formed like shown below.

Reviewer: ```suggestion
When nodes are connected to one other, a linked list is formed, as shown below:
```

![Singular linked list](https://raw.githubusercontent.com/Codecademy/docs/main/media/singular-linked-list.png)

> So the linked consist to parts:

Reviewer: ```suggestion
A linked list consists of two key components:
```

> - `head`: The first node of the linked list.
> - `tail`: The last node of the linked list.

Reviewer: ```suggestion
- `Head`: The first node in the linked list.
- `Tail`: The last node in the linked list, which points to `NULL` in a singly linked list.
```
